Error: `panic: cp flag redefined: help` from `s5cmd run`

I'm using `s5cmd run` to issue a large number of `cp` commands in parallel. I intermittently see runs fail with this stack trace:

```
panic: cp flag redefined: help

goroutine 23 [running]:
flag.(*FlagSet).Var(0xc0004000c0, 0xd860e0, 0xc000422338, 0xc000422318, 0x4, 0xb5eaee, 0x9)
	/opt/hostedtoolcache/go/1.14.12/x64/src/flag/flag.go:851 +0x4b8
flag.(*FlagSet).BoolVar(...)
	/opt/hostedtoolcache/go/1.14.12/x64/src/flag/flag.go:624
flag.(*FlagSet).Bool(0xc0004000c0, 0xc000422318, 0x4, 0x0, 0xb5eaee, 0x9, 0x2)
	/opt/hostedtoolcache/go/1.14.12/x64/src/flag/flag.go:637 +0x8a
github.com/urfave/cli/v2.(*BoolFlag).Apply(0x11d07a0, 0xc0004000c0, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/flag_bool.go:81 +0x201
github.com/urfave/cli/v2.flagSet(0xb59958, 0x2, 0xc0001c29a0, 0xd, 0x16, 0x203000, 0x203000, 0xc00012ada8)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/flag.go:125 +0x93
github.com/urfave/cli/v2.(*Command).newFlagSet(...)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:173
github.com/urfave/cli/v2.(*Command).parseFlags(0x11d1b40, 0xd90f20, 0xc000416000, 0xd8e800, 0x11d07a0, 0xb47500, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:181 +0x62
github.com/urfave/cli/v2.(*Command).Run(0x11d1b40, 0xc0001b14c0, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:106 +0x100
github.com/peak/s5cmd/command.glob..func19.2(0x0, 0x0)
	/home/runner/work/s5cmd/s5cmd/command/run.go:123 +0x1d1
github.com/peak/s5cmd/parallel.(*Manager).Run.func1(0xc0001b9f00, 0xc000382b60, 0xc000369ec0)
	/home/runner/work/s5cmd/s5cmd/parallel/parallel.go:57 +0x75
created by github.com/peak/s5cmd/parallel.(*Manager).Run
	/home/runner/work/s5cmd/s5cmd/parallel/parallel.go:53 +0xa9
```

I speculate that the bug is a concurrency issue where multiple goroutines are executing a `cp` in parallel and somehow end up trying to populate the flags concurrently; but I haven't isolated the exact issue.

Just saw this one on CI: https://github.com/peak/s5cmd/pull/270/checks?check_run_id=2962063863

```
Run make test
  make test
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    GOROOT: C:\hostedtoolcache\windows\go\1.16.5\x64
?   	github.com/peak/s5cmd	[no test files]
ok  	github.com/peak/s5cmd/command	0.124s
--- FAIL: TestRunFromFileJSON (0.06s)
    run_test.go:162: assertion failed: 
        Command:  C:\Users\RUNNER~1\AppData\Local\Temp\621052015\s5cmd.exe --endpoint-url http://127.0.0.1:50060 --json run C:\Users\RUNNER~1\AppData\Local\Temp\prefix-640674345
        ExitCode: 2
        Error:    exit status 2
        Stdout:   
        Stderr:   ls flag redefined: help
        panic: ls flag redefined: help
        
        goroutine 10 [running]:
        flag.(*FlagSet).Var(0xc0002801e0, 0x1b4a1a0, 0xc000286178, 0xc000286158, 0x4, 0x1a0a315, 0x9)
        	C:/hostedtoolcache/windows/go/1.16.5/x64/src/flag/flag.go:871 +0x48b
        flag.(*FlagSet).BoolVar(...)
        	C:/hostedtoolcache/windows/go/1.16.5/x64/src/flag/flag.go:630
        flag.(*FlagSet).Bool(0xc0002801e0, 0xc000286158, 0x4, 0x0, 0x1a0a315, 0x9, 0x2)
        	C:/hostedtoolcache/windows/go/1.16.5/x64/src/flag/flag.go:643 +0x91
        github.com/urfave/cli/v2.(*BoolFlag).Apply(0x1e8df40, 0xc0002801e0, 0x0, 0x0)
        	D:/a/s5cmd/s5cmd/vendor/github.com/urfave/cli/v2/flag_bool.go:81 +0x219
        github.com/urfave/cli/v2.flagSet(0x1a050c0, 0x2, 0xc000054d80, 0x5, 0x6, 0x1ee0d88, 0xc000046c00, 0x0)
        	D:/a/s5cmd/s5cmd/vendor/github.com/urfave/cli/v2/flag.go:125 +0xe6
        github.com/urfave/cli/v2.(*Command).newFlagSet(...)
        	D:/a/s5cmd/s5cmd/vendor/github.com/urfave/cli/v2/command.go:173
        github.com/urfave/cli/v2.(*Command).parseFlags(0x1e8f620, 0x1b52560, 0xc000290000, 0x1b50400, 0x1e8df40, 0x100, 0x110)
        	D:/a/s5cmd/s5cmd/vendor/github.com/urfave/cli/v2/command.go:181 +0x69
        github.com/urfave/cli/v2.(*Command).Run(0x1e8f620, 0xc00028c000, 0x0, 0x0)
        	D:/a/s5cmd/s5cmd/vendor/github.com/urfave/cli/v2/command.go:106 +0x107
        github.com/peak/s5cmd/command.glob..func20.2(0x0, 0x0)
        	D:/a/s5cmd/s5cmd/command/run.go:123 +0x21e
        github.com/peak/s5cmd/parallel.(*Manager).Run.func1(0xc000004360, 0xc000248b90, 0xc000223ec0)
        	D:/a/s5cmd/s5cmd/parallel/parallel.go:57 +0x7c
        created by github.com/peak/s5cmd/parallel.(*Manager).Run
        	D:/a/s5cmd/s5cmd/parallel/parallel.go:53 +0xb5
        
        
        Failures:
        ExitCode was 2 expected 0
        Expected no error
FAIL
FAIL	github.com/peak/s5cmd/e2e	21.449s
?   	github.com/peak/s5cmd/error	[no test files]
?   	github.com/peak/s5cmd/log	[no test files]
?   	github.com/peak/s5cmd/log/stat	[no test files]
?   	github.com/peak/s5cmd/parallel	[no test files]
?   	github.com/peak/s5cmd/parallel/fdlimit	[no test files]
ok  	github.com/peak/s5cmd/storage	0.130s
ok  	github.com/peak/s5cmd/storage/url	0.026s
?   	github.com/peak/s5cmd/strutil	[no test files]
?   	github.com/peak/s5cmd/version	[no test files]
FAIL
mingw32-make: *** [Makefile:16: test] Error 1
Error: Process completed with exit code 1.

```

--- a/command/run.go
+++ b/command/run.go
@@ -120,7 +120,9 @@ var runCommand = &cli.Command{
 				}
 
 				ctx := cli.NewContext(app, flagset, c)
-				return cmd.Run(ctx)
+
+				safecmd := *cmd
+				return safecmd.Run(ctx)
 			}
 
 			pm.Run(fn, waiter)

--- a/command/run.go
+++ b/command/run.go
@@ -106,8 +106,8 @@ var runCommand = &cli.Command{
 			fn := func() error {
 				subcmd := fields[0]
 
-				cmd := app.Command(subcmd)
-				if cmd == nil {
+				cmdptr := app.Command(subcmd)
+				if cmdptr == nil {
 					err := fmt.Errorf("%q command (line: %v) not found", subcmd, lineno)
 					printError(givenCommand(c), c.Command.Name, err)
 					return nil
@@ -121,8 +121,10 @@ var runCommand = &cli.Command{
 
 				ctx := cli.NewContext(app, flagset, c)
 
-				safecmd := *cmd
-				return safecmd.Run(ctx)
+				// Use the dereferenced value to avoid urfave/cli package to
+				// mutate the command. Ref: https://github.com/peak/s5cmd/issues/301
+				cmd := *cmdptr
+				return cmd.Run(ctx)
 			}
 
 			pm.Run(fn, waiter)

--- a/e2e/run_test.go
+++ b/e2e/run_test.go
@@ -282,3 +282,35 @@ func TestRunDryRun(t *testing.T) {
 	// ensure no side effect for remove operation
 	assert.Assert(t, ensureS3Object(s3client, bucket, files[2], "content"))
 }
+
+func TestRunFixDataRace_Issue301(t *testing.T) {
+	t.Parallel()
+
+	bucket := s3BucketFromTestName(t)
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, bucket)
+	putFile(t, s3client, bucket, "file1.txt", "content")
+
+	// The fix is applicable to the run command, where there should be at least
+	// 2 of the same command given as input.
+	input := strings.NewReader(
+		strings.Join([]string{
+			fmt.Sprintf("ls s3://%v/file1.txt", bucket),
+			fmt.Sprintf("ls s3://%v/file1.txt", bucket),
+		}, "\n"),
+	)
+	cmd := s5cmd("run")
+	result := icmd.RunCmd(cmd, icmd.WithStdin(input))
+
+	result.Assert(t, icmd.Success)
+
+	assertLines(t, result.Stdout(), map[int]compareFunc{
+		0: suffix("file1.txt"),
+		1: suffix("file1.txt"),
+	}, sortInput(true))
+
+	assertLines(t, result.Stderr(), map[int]compareFunc{})
+}

